### DISCUSSION TITLE
What does $rectorConfig->phpVersion(); do?

### DISCUSSION THREAD

**Discussion by ziming** [rectorphp/rector#7351] [Q&A]
Hi,

i read the readme but this still confuses me. Say my computer is on PHP 8.1 and i have an open source project that support minimum PHP 7.3

in my rector.php file

do i define 7.3 or do i define 8.1?

$rectorConfig->phpVersion(PhpVersion::PHP_73);

or

$rectorConfig->phpVersion(PhpVersion::PHP_81);

**Comment by acentrist** [comment]
You should use symbolic links.
By doing so, you can manage the development version separately from other working directories.

**Reply by TomasVotruba** [reply]
Hi,

this set PHP version syntax level for the Rector rules. E.g. if you set:

```php
$rectorConfig->phpVersion(PhpVersion::PHP_74);
```

If there is a rule that completes type to a property, it will use the PHP 7.4 typed property:

```diff
-private $name;
+private string $name;
```

But if you set PHP 7.3:

```php
$rectorConfig->phpVersion(PhpVersion::PHP_73);
```

Nothing will be changed there :) 

Usually this version is resolved based on your `composer.json`, where the lowest PHP version supported is defined.

**Reply by acentrist** [reply]
wow, thanks!

**Reply by ziming** [reply]
Thank you

**Reply by TomasVotruba** [reply]
You're welcome :slightly_smiling_face:
